navbar.ts

```diff
@@ -0,0 +1,433 @@
+  @property({ type: String }) canvascls = "canvas-body";
+
+  noParentAnchors: NodeType[] = [];
+  noParentContianer: HTMLAnchorElement[] = [];
+  parentContainer: { [key: string]: HTMLAnchorElement[] } = {};
+  anchorElements: HTMLAnchorElement[];
+  menu: HTMLElement;
+  menuElement: HTMLAnchorElement;
+  closeElement: HTMLAnchorElement;
+  backElement: HTMLAnchorElement;
+  currentPath: HTMLAnchorElement[] = [];
+  change: boolean = false;
+  overlayContainer: { [key: string]: HTMLElement[] } = {};
+  activeClickOutsideListener: ((event: MouseEvent) => void) | null = null;
+  navbar: HTMLElement;
+
+  override createRenderRoot() {
+    return this;
+  }
+  override firstUpdated() {
+    this.navbar = document.querySelector("aalam-navbar") as HTMLElement;
+
+    const original = this.querySelectorAll(
+      this.attr ? `[${this.attr}]` : "a"
+    ) as NodeListOf<HTMLAnchorElement>;
+    this.anchorElements = Array.from(original).map(
+      (anchor) => anchor.cloneNode(true) as HTMLAnchorElement
+    );
+
+    const lengthOfInput = this.anchorElements.length;
+
+    let i = 0;
+    while (i < lengthOfInput) {
+      const anchor = this.anchorElements[i];
+      const attrValue = anchor.getAttribute(this.attr) || "";
+      const attributes = this.parseAttributes(attrValue);
+      const attrName = attributes.nm;
+
+      if (!attributes.pnt) {
+        this.noParentAnchors.push({ elementName: attrName, children: [] });
+        if (!(attrName === "menu") && !(attrName == "back"))
+          this.noParentContianer.push(anchor);
+      } else {
+        if (!this.parentContainer[attributes.pnt]) {
+          this.parentContainer[attributes.pnt] = [];
+        }
+        this.parentContainer[attributes.pnt].push(anchor);
+      }
+      i++;
+    }
+    
+    const toggle = this.querySelector(this.togglesel);
+    if (toggle) {
+      const originalMenu = toggle as HTMLAnchorElement;
+      this.menuElement = originalMenu.cloneNode(true) as HTMLAnchorElement;
+      originalMenu.style.display = "none";
+      document.body.appendChild(this.menuElement);
+    }
+
+    this.menuElement.style.display = "block";
+    this.menuElement.style.width = "100%";
+    this.menuElement.style.background = "blue";
+
+    let temps = document.getElementsByTagName("template");
+
+    Array.from(temps).forEach((template) => {
+      if (template.getAttribute("data-type") === "back") {
+        const newContent = template.content.cloneNode(true) as DocumentFragment;
+        this.backElement = document.createElement("a");
+        this.backElement.appendChild(newContent.cloneNode(true));
+      } else if (template.getAttribute("data-type") === "close") {
+        const newContent = template.content.cloneNode(true) as DocumentFragment;
+        this.closeElement = document.createElement("a");
+        this.closeElement.appendChild(newContent.cloneNode(true));
+      }
+    });
+
+    this.backElement.id = "back";
+    this.closeElement.id = "close";

```

Comment:
Its better to avoid adding these ids, as this might conflict with the existing element within the other parts of the documet. Also, we can manage very well with the use of ids in this case.